compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/Focusable.kt

```diff
@@ -141,7 +141,8 @@ private val FocusableInNonTouchModeElement =
 
         override fun update(node: FocusableInNonTouchMode) {}
 
-        override fun hashCode(): Int = System.identityHashCode(this)
+        private val arbitraryHashCode: Int = 0x70f4de02 // Just a randomly generated constant
+        override fun hashCode(): Int = arbitraryHashCode
```

Comment:
```suggestion
        override fun hashCode(): Int = identityHashCode()
```
and add an import `androidx.compose.runtime.identityHashCode`

Reply:
Ah, it is internal. Better than to copy it to foundation as well

Reply:
Added `@InternalComposeApi` to identityHashCode and reuse in from runtime module

Reply:
In similar cases we copy-pasted the code to other modules, even though reusing from runtime was possible.

I don't remember exactly why we have this rule. Maybe Igor will recall it.

Reply:
if we make it public even with InternalComposeApi, we have to keep binary compatibility between different versions of runtime/foundation

Reply:
Is it possible only when a user manually forces an older runtime version with a new foundation?

For the future versions, I see the reason, that despite an annotation, we cant remove that function (if we decided to delete it) to not break the 3rd party libraries.

Reply:
I think that `identityHashCode` will be easy to maintain.

In this PR have actual's in 4 sourceSet's:
 - jvm (Desktop+Android)
 - js
 - wasm
 - native
 

And remove redundant parts:
 In 2 another modules we and 5 sourceSets:
  - android
  - desktop
  - js
  - wasm
  - native

Reply:
Will be easier to have it only in `runtime` module

Reply:
This function is pretty stable though, we can make it public @InternalComposeApi.

This change can be also rejected during upstreaming, but in this case we just move it to skikoMain

Reply:
Approve?